### PR TITLE
api.models: implement new Node.state values

### PR DESCRIPTION
**Pull request by gctucker**
Replace Node.status with Node.state to make it clear it's about the
state of the Node and not about any pass/fail result.  Update the list
of states to match the new specs.

Signed-off-by: Guillaume Tucker <guillaume.tucker@collabora.com>

### CODE REVIEW

**Comment by gctucker** [comment]
Adding `staging-skip` until the pipeline services have been updated to use the new state names.  Also existing nodes in the database will have to be removed (or migrated...).

File: test/test_node_handler.py [hunk 1]
```diff
@@ -351,7 +351,7 @@ def test_get_root_node_endpoint(mock_db_find_by_id, mock_init_sub_id):
             name="checkout",
             revision=revision_obj,
             parent="61bda8f2eb1a63d2b7152410",
-            status="pending",
+            state="pending",
```
Comment (JenySadadia) [inline test/test_node_handler.py:354]: Although the state value doesn't make much of a difference in these tests, the node state value should be changed to the default one.
```suggestion
            state="running",
```
This should be applied to all the nodes in tests.

Comment (gctucker) [reply test/test_node_handler.py:354]: Why does it need to be the default value?

Comment (JenySadadia) [reply test/test_node_handler.py:354]: I think, If we are setting the random value to node state for tests, the default one should be preferred.
I don't see why we should set it to 'pending'.

Comment (gctucker) [reply test/test_node_handler.py:354]: Well I think we should be testing all the values in principle, and I don't see why the default one is preferred to other ones.  The only reason we might test the default one is if we don't set a value we can check the default one is used when the node is created.

**Review by JenySadadia — APPROVED** [review]

File: test/test_node_handler.py (hunk 1 above)
Comment (JenySadadia) [reply test/test_node_handler.py:354]: Makes sense.

**Pull request merged**